src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -110,7 +111,7 @@ const buildIterableFunctionLoop: LoopBuilder = makeForLoopBuilder((state, name,
 	return exp;
 });
 
-const buildIterableFunctionLuaTupleLoop: LoopBuilder = (state, statements, name, exp) => {
+const buildIterableFunctionLuaTupleLoop: (type: ts.Type) => LoopBuilder = type => (state, statements, name, exp) => {
```

Comment:
This looks good, but I'd prefer to keep the old code as a fallback case. We cannot infer how many return values there will be for stuff like `IterableFunction<LuaTuple<Array<number>>>`